src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,152 @@
 import React from 'react';
+import { getTodos } from '../../api';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+enum Status {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+type Props = {
+  selectedUserId: number;
+  selectUser: (userId: number) => void;
+};
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
+type State = {
+  todos: Todo[];
+  status: Status;
+  query: string;
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    todos: [],
+    status: 'all' as Status,
+    query: '',
+  };
+
+  async componentDidMount() {
+    const todos = await getTodos();
+
+    this.setState({ todos });
+  }
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ query: event.target.value });
+  };
+
+  handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ status: event.target.value as Status });
+  };
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+  randomSort = () => {
+    this.setState((prevState) => ({
+      todos: prevState.todos.sort(() => Math.random() - 0.5),
+    }));
+  };
+
+  filterTodos = () => {
+    const { todos, status, query } = this.state;
+    const filteredTodos = todos.filter(todo => {
+      switch (status) {
+        case Status.active:
+          return !todo.completed;
+        case Status.completed:
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+
+    return filteredTodos.filter(todo => (
+      todo.title
+      && todo.title.toLowerCase().includes(query.toLowerCase())));
```

Comment:
better make a separate variable for query.toLowerCase() not to convert it on every array iteration